prometheus/src/identity_info.rs

```diff
@@ -0,0 +1,127 @@
+use solana_config_program::ConfigKeys;
+use solana_sdk::pubkey::Pubkey;
+use solana_vote_program::vote_state::VoteState;
+
+use bincode;
+use serde::Deserialize;
+use serde_json;
+use solana_runtime::accounts_index::ScanConfig;
+use solana_runtime::bank_forks::BankForks;
+use solana_sdk::account::ReadableAccount;
+use solana_sdk::transaction_context::TransactionAccount;
+use std::collections::HashMap;
+use std::sync::RwLock;
+use std::{collections::HashSet, sync::Arc};
+
+/// ValidatorInfo represents selected fields from the config account data.
+#[derive(Debug, Default, Deserialize, Clone, Eq, PartialEq)]
+pub struct ValidatorInfo {
+    pub name: String,
+}
+
+pub type IdentityInfoMap = HashMap<Pubkey, ValidatorInfo>;
+
+pub fn map_vote_identity_to_info(
+    bank_forks: &Arc<RwLock<BankForks>>,
+    vote_pubkeys: &Arc<HashSet<Pubkey>>,
+) -> IdentityInfoMap {
+    use solana_sdk::config::program as config_program;
+
+    let bank = bank_forks.read().unwrap().working_bank();
```

Comment:
Note to self: `RwLock::read()` only returns `Err` if the lock was poisoned (which happens when a thread panics while holding the lock). This is a well-known design flaw, Rust should have just made that case panic as well. So the unwrap here is fine.